lib/Serialization/Serialization.cpp

```diff
@@ -3499,6 +3500,16 @@ class Serializer::DeclSerializer : public DeclVisitor<DeclSerializer> {
   }
 
   void visitAccessorDecl(const AccessorDecl *fn) {
+    // Accessor synthesis and type checking is now sufficiently lazy that
+    // we might have unvalidated accessors in a primary file.
+    //
+    // FIXME: Once accessor synthesis and getInterfaceType() itself are
+    // request-ified this goes away.
+    if (!fn->hasValidSignature()) {
+      S.M->getASTContext().getLazyResolver()->resolveDeclSignature(
```

Comment:
Can you assert that the accessor is synthesized in this case? We definitely do not want any explicitly-written accessors to be skipped.

Reply:
No problem. However do keep in mind this whole 'if' check will go away at some point.